### DISCUSSION TITLE
Guidance for React Auth UI component ("@supabase/auth-ui-react")

### DISCUSSION THREAD

**Discussion by user72356** [supabase/supabase#9013] [Questions]
I followed the Auth UI React guide at https://supabase.com/docs/guides/auth/auth-helpers/auth-ui and I managed to get the Auth component to render on the page. Apparently I can register and log in, as evidenced by the log on the admin UI. 

Unfortunately the guide doesn't document how to integrate the component within the wider app, such as how to use it to protect the app or parts of it, how to display specific content when the login succeeds/fails, when registration succeeds/fails, etc. It seems like the guide is half finished at best. 

Previously I used the AWS Amplify Auth component and I had to wrap it around my App component, and everything was handled automatically, including showing confirmation screens after login, etc. I'm not sure how to do that here??

**Comment by kyleknighted** [comment]
Not 100% sure about the library, could use this (or concepts from it) - https://react-supabase.vercel.app/documentation/provider

In short, create Context Provider that wraps your App and call a custom hook for the `user` status. 

You could also use a middleware, like Redux, and during a dispatch update, validate that the user present.
